src/plugins/custom_import_map/public/components/vector_upload_options.tsx

```diff
@@ -0,0 +1,339 @@
+  const INDEX_NAME_NOT_ENDS_WITH_CHECK = /.*-map$/;
+  const MAX_LENGTH_OF_INDEX_NAME = 250;
+
+  const options = [
+    { value: 'geo_shape', text: 'Geo shape', name: 'geo_shape' },
+    { value: 'geo_point', text: 'Geo point', name: 'geo_point' },
+  ];
+
+  const [value, setValue] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const onSelectChange = (e) => {
+    setSelectValue(e.target.value);
+  };
+
+  const [selectValue, setSelectValue] = useState('');
+  const [fileContent, setFileContent] = useState();
+
+  const onTextChange = (e) => {
+    setValue(e.target.value);
+  };
+
+  const onChange = (files) => {
+    if (files[0]) {
+      validateFileSize(files);
+      setFileContent(files);
+    }
+  };
+
+  const fetchElementByName = (elementName: string) => {
+    return document.getElementsByName(elementName)[0];
+  };
+
+  const validateIndexName = (typedIndexName: string, isIndexNameWithSuffix: boolean) => {
+    let error = '';
+    const errorIndexNameDiv = fetchElementByName('errorIndexName');
+
+    // check for presence of index name entered by the user
+    if (!typedIndexName) {
+      error = 'Required';
+    } else {
+      // check for restriction on length of the index name
+      if (MAX_LENGTH_OF_INDEX_NAME < typedIndexName.length) {
+        error += ' Map name should be less than ' + MAX_LENGTH_OF_INDEX_NAME + ' characters.\n';
+      }
+
+      // check for restriction on the usage of upper case characters in the index name
+      if (INDEX_NAME_UPPERCASE_CHECK.test(typedIndexName)) {
+        error += ' Upper case letters are not allowed.\n';
+      }
+
+      // check for restriction on the usage of special characters in the index name
+      if (INDEX_NAME_SPECIAL_CHARACTERS_CHECK.test(typedIndexName)) {
+        error += ' Special characters are not allowed.\n';
+      }
+
+      // check for restriction on the usage of characters at the beginning in the index name
+      if (
+        INDEX_NAME_NOT_BEGINS_WITH_CHECK.test(typedIndexName) ||
+        !INDEX_NAME_BEGINS_WITH_CHECK.test(typedIndexName)
+      ) {
+        error += " Map name can't start with + , _ , - or . It should start with a-z.\n";
+      }
+
+      // check for restriction on the usage of -map in map name if entered by the user
+      if (!isIndexNameWithSuffix && INDEX_NAME_NOT_ENDS_WITH_CHECK.test(typedIndexName)) {
+        error += " Map name can't end with -map.\n";
+      }
+    }
+
+    if (error) {
+      errorIndexNameDiv.textContent = error;
+      setLoading(false);
+      return false;
+    }
+    errorIndexNameDiv.textContent = '';
+    return true;
+  };
+
+  const validateFileSize = async (files) => {
+    // check if the file size is permitted
+    if (FILE_PAYLOAD_SIZE < files[0].size) {
+      notifications.toasts.addDanger(
+        'File size should be less than ' + FILE_PAYLOAD_SIZE_IN_MB + ' MB.'
+      );
+      setLoading(false);
+      return false;
+    } else if (files[0].size === 0) {
```

Comment:
You don’t need else since previous if returns at end of block

Reply:
modified this, thanks!